osutil/buildid.go

```diff
@@ -36,18 +36,16 @@ type elfNoteHeader struct {
 	Type   uint32
 }
 
-// ReadBuildID returns the GNU build ID note of the provided ELF executable.
-// The ErrNoBuildID error is returned when one is not found.
-//
-// Observed Go binaries presented one when built with:
-//
-//      go build -buildmode=pie
-//
-// See details at http://fedoraproject.org/wiki/Releases/FeatureBuildId
+// ReadBuildID will return the GNU build-id if available and else the
+// GO build-id (the go-buildid is only available
 func ReadBuildID(fname string) (string, error) {
-	const ELF_NOTE_GNU = "GNU\x00"
-	const NT_GNU_BUILD_ID uint32 = 3
+	if buildId, err := readGenericBuildID(fname, "GNU\x00", 3); err == nil {
+		return buildId, nil
+	}
+	return readGenericBuildID(fname, "Go\x00\x00", 4)
```

Comment:
Can you provide two `const` definitions for the 3 and 4 magic values here?

As a separate question, why does go note name use two NUL bytes? To have the same length as the GNU note?